src/NUnitTestAdapter/NUnitTestAdapter.cs

```diff
@@ -203,9 +203,15 @@ private TestPackage CreateTestPackage(string assemblyName)
             {
                 package.Settings[PackageSettings.DomainUsage] = "Single";
             }
-
-            // Force truncation of string arguments to test cases
-            package.Settings[PackageSettings.DefaultTestNamePattern] = "{m}{a:40}";
+            if (Settings.DefaultTestNamePattern != null)
+            {
+                package.Settings[PackageSettings.DefaultTestNamePattern] = Settings.DefaultTestNamePattern;
+            }
+            else
+            {
+                // Force truncation of string arguments to test cases
+                package.Settings[PackageSettings.DefaultTestNamePattern] = "{m}{a:40}";
```

Comment:
First I thought it could confuse people, so I thought it best to just do one change, and not multiple, but would it really confuse?   If you hit the limit, then the test results duplicates, so you would see different numbers, but we could just state that in the release notes.  A default limit of 80 as you say could be easier on people.   I am also checking with MS if there are any real limits in VS on this, waiting for answer on that.